src/graph/unit_graph.cc

```diff
@@ -851,24 +974,41 @@ HeteroSubgraph UnitGraph::EdgeSubgraph(
 }
 
 HeteroGraphPtr UnitGraph::CreateFromCOO(
-    int64_t num_vtypes, int64_t num_src, int64_t num_dst, IdArray row, IdArray col) {
+    int64_t num_vtypes, int64_t num_src, int64_t num_dst, IdArray row, IdArray col,
+    SparseFormat restrict_format) {
   CHECK(num_vtypes == 1 || num_vtypes == 2);
   if (num_vtypes == 1)
     CHECK_EQ(num_src, num_dst);
   auto mg = CreateUnitGraphMetaGraph(num_vtypes);
   COOPtr coo(new COO(mg, num_src, num_dst, row, col));
-  return HeteroGraphPtr(new UnitGraph(mg, nullptr, nullptr, coo));
+
+  return HeteroGraphPtr(
+      new UnitGraph(mg, nullptr, nullptr, coo, restrict_format));
 }
 
 HeteroGraphPtr UnitGraph::CreateFromCSR(
     int64_t num_vtypes, int64_t num_src, int64_t num_dst,
-    IdArray indptr, IdArray indices, IdArray edge_ids) {
+    IdArray indptr, IdArray indices, IdArray edge_ids, SparseFormat restrict_format) {
   CHECK(num_vtypes == 1 || num_vtypes == 2);
   if (num_vtypes == 1)
     CHECK_EQ(num_src, num_dst);
   auto mg = CreateUnitGraphMetaGraph(num_vtypes);
   CSRPtr csr(new CSR(mg, num_src, num_dst, indptr, indices, edge_ids));
-  return HeteroGraphPtr(new UnitGraph(mg, nullptr, csr, nullptr));
+  return HeteroGraphPtr(new UnitGraph(mg, nullptr, csr, nullptr, restrict_format));
+}
+
+HeteroGraphPtr CreateUnitGraphFromCOO(
+    int64_t num_vtypes, int64_t num_src, int64_t num_dst, IdArray row, IdArray col,
+    SparseFormat restrict_format) {
+  return UnitGraph::CreateFromCOO(num_vtypes, num_src, num_dst, row, col, restrict_format);
+}
+
+HeteroGraphPtr CreateUnitGraphFromCSR(
+    int64_t num_vtypes, int64_t num_src, int64_t num_dst,
+    IdArray indptr, IdArray indices, IdArray edge_ids,
+    SparseFormat restrict_format) {
+  return UnitGraph::CreateFromCSR(
+      num_vtypes, num_src, num_dst, indptr, indices, edge_ids, restrict_format);
```

Comment:
Maybe I missed sth here. Why need these two creator functions?

Reply:
Should we export these creation functions?

Reply:
I c. You could decide.